pkg/ring/lifecycler.go

```diff
@@ -696,6 +696,13 @@ func (i *Lifecycler) initRing(ctx context.Context) error {
 			return ringDesc, true, nil
 		}
 
+		if i.cfg.HeartbeatPeriod == 0 && i.Addr != instanceDesc.Addr {
+			// Update the address if it has changed
+			instanceDesc.Addr = i.Addr
```

Comment:
I think we can have this line before line 692 and lets the existing deep equal check take care if the Ip changed?

Setting the `instanceDesc.Addr = i.Addr` should be a noop in case the IP did not change, if it changed, the check on https://github.com/cortexproject/cortex/pull/6271/files#diff-4d46760513b4f0a55e4cebb89e6a055859204f00ca50220cd42b5191b018b152R692 should take care of updating the ring.

Reply:
thanks!